packages/react-components/src/queries/useHistoricalAssetPropertyValues/sync.ts

```diff
@@ -0,0 +1,77 @@
+import { useEffect } from 'react';
+import { StateCreator, create } from 'zustand';
+import { devtools } from 'zustand/middleware';
+
+export interface HistoricalAssetPropertyValueSync {
+  enabled: Record<number, boolean>;
+}
+
+export interface HistoricalAssetPropertyValueSyncState
+  extends HistoricalAssetPropertyValueSync {
+  enable: (refreshRate: number) => void;
+  disable: (refreshRate: number) => void;
+}
+
+export const createHistoricalAssetPropertyValueSyncSlice: StateCreator<
+  HistoricalAssetPropertyValueSyncState
+> = (set) => ({
+  enabled: {},
+  enable: (refreshRate: number) =>
+    set((state) => ({
+      ...state,
+      enabled: {
+        ...state.enabled,
+        [refreshRate]: true,
+      },
+    })),
+  disable: (refreshRate: number) =>
+    set((state) => ({
+      ...state,
+      enabled: {
+        ...state.enabled,
+        [refreshRate]: false,
+      },
+    })),
+});
+
+export const useHistoricalAssetPropertyValueSyncStore =
```

Comment:
Can we reuse the same sync store between the Latest and Historical syncs? They are identical

Reply:
Ya, moved this into a separate package.